march_rqt_gait_generator/test/nosetests/modifiable_joint_trajectory_test.py

```diff
@@ -0,0 +1,259 @@
+import copy
+import unittest
+
+from mock import Mock
+
+from march_rqt_gait_generator.model.modifiable_joint_trajectory import ModifiableJointTrajectory
+from march_rqt_gait_generator.model.modifiable_setpoint import ModifiableSetpoint
+from march_shared_classes.gait.limits import Limits
+
+
+class ModifiableJointTrajectoryTest(unittest.TestCase):
+    def setUp(self):
+        self.gait_generator = Mock()
+        self.joint_name = 'test_joint'
+        self.limits = Limits(-1, 1, 2)
+        self.duration = 2.0
+        self.times = [0, self.duration / 2.0, self.duration]
+        self.setpoints = [ModifiableSetpoint(t, 0.5 * t, t) for t in self.times]
+        self.joint_trajectory = ModifiableJointTrajectory(self.joint_name, self.limits, copy.deepcopy(self.setpoints),
```

Comment:
deepcopy seems redundant since the setup is called every testing method.

Reply:
Ah, that was necessary because in a few of the tests I wanted to modify self.setpoints without modifying self.joint_trajectory. Hence, a deepcopy is necessary.